File: Practica2/python/CirculoDibujo.py
import tkinter as tk

class Punto:
    def __init__(self, x: int, y: int):
        self.x = x
        self.y = y

class Circulo:
    def __init__(self, centro: Punto, radio: float):
        self.centro = centro
        self.radio = radio

    def dibujaCirculo(self):
        ventana = tk.Tk()
        ventana.title("Dibujar Círculo")
        ventana.geometry("420x440")
        ventana.resizable(False, False)

        canvas = tk.Canvas(ventana, width=600, height=600, bg="white")
        canvas.pack()

        x1 = (self.centro.x * 10) - (self.radio * 10)
        y1 = (self.centro.y * 10) - (self.radio * 10)
        x2 = (self.centro.x * 10) + (self.radio * 10)
        y2 = (self.centro.y * 10) + (self.radio * 10)

        canvas.create_oval(x1, y1, x2, y2, outline="pink", width=2)

        ventana.mainloop()


if __name__ == "__main__":
    centro = Punto(20, 20)
    radio = 7.5
    circulo = Circulo(centro, radio)
    circulo.dibujaCirculo()

File: Practica2/Java/CirculoDibujo.java
package Practica2.Java;
import javax.swing.*;
import java.awt.*;


class Punto {
    public int x, y;

    public Punto(int x, int y) {
        this.x = x;
        this.y = y;
    }
}

class Circulo extends JPanel {
    Punto centro;
    float radio;


    public Circulo(Punto centro, float radio) {
        this.centro = centro;
        this.radio = radio;
        setPreferredSize(new Dimension(600, 600)); 
    }

    @Override
    protected void paintComponent(Graphics g) {
        super.paintComponent(g);
        g.setColor(Color.PINK);
        g.drawOval(centro.x * 10, centro.y * 10, (int) (2 * radio * 10), (int) (2 * radio * 10)); 
    }

    public void dibujaCirculo() {
        JFrame frame = new JFrame("Dibujar Círculo");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setSize(620, 640);
        frame.setResizable(false);
        frame.setLocationRelativeTo(null); 
        frame.add(this);
        frame.setVisible(true);
    }
}


public class CirculoDibujo {
    public static void main(String[] args) {
        Punto centro = new Punto(20, 20); 
        float radio = 7.5f;
        Circulo circulo = new Circulo(centro, radio);
        circulo.dibujaCirculo();
    }
}


